### ISSUE TITLE
hvd.allreduce() not working when placed in TF graph

### ISSUE THREAD

**Issue by rahul003** [uber/horovod#375]
Hello,
I'm trying to shard my validation dataset across horovod workers and use allreduce to sum them up. 
But the allreduce doesn't seem to work. It gives me the same answer with or without the allreduce. Any idea what could be wrong? Can I use allreduce like this?
```
with tf.device(None): # Allow fallback to CPU if no GPU support for these ops
                accuracy = tf.metrics.accuracy(
                    labels=labels, predictions=predicted_classes)
                # tf.summary.scalar('val-accuracy', accuracy[1])
                top5acc = tf.metrics.mean(
                    tf.cast(tf.nn.in_top_k(logits, labels, 5), tf.float32))
                accuracy = (accuracy[0], 100. * hvd.allreduce(accuracy[1]))
                top5acc = (top5acc[0], 100. * hvd.allreduce(top5acc[1]))
                tf.summary.scalar('accuracy', accuracy[1])
                tf.summary.scalar('top5-accuracy', top5acc[1])
                metrics = {'val-top1acc': accuracy, 'val-top5acc': top5acc}
            return tf.estimator.EstimatorSpec(
                mode, loss=loss, eval_metric_ops=metrics)
```

**Comment by tgaddair** [comment]
Hey @rahul003, that looks like it should work at a glance.  Are you certain the values are different?  What happens if you run with:

```
hvd.allreduce(accuracy[1], average=False)
```

Also, can you print out `hvd.size()` to verify that you have more than one worker?

**Comment by rahul003** [comment]
Ah, sorry it was my mistake. The accuracy value is accuracy[0] not accuracy[1]. Second value is the update_op.
